staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -1049,6 +1063,30 @@ func (cfgCtlr *configController) maybeReapReadLocked(plName string, plState *pri
 	cfgCtlr.configQueue.Add(0)
 }
 
+// PriorityLevelMaxSeats returns the maximum seats that a request should occupy
+// for a given priority level. See priorityLevelMaxSeatsLocked for more details.
+func (cfgCtlr *configController) PriorityLevelMaxSeats(plName string) uint64 {
+	// This function should only be invoked when work estimator runs in startRequest,
+	// where cgfCtlr's lock is already acquired.
```

Comment:
This one is tricky:
1) I would rename this function to PriorityLevelMaxSeatsLocked to make that clear
2) Do we have any way to enforce that (do we know if estimator is always called under lock)

The alternative would be change the interface of estimator to take maxSeats as an additiona argument...

Reply:
I didn't name this `PriorityLevelMaxSeatsLocked` cause I didn't want to leak the locking implementation details in the exported interface. I think the risk is lower now that `PriorityLevelMaxSeats()` returns MaxSeatsFunc -- but not sure we have any way to enforce it at the moment. 

> (do we know if estimator is always called under lock

As far as I can tell, workEstimator is only passed into startRequest in the handler: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_filter.go#L161

Reply:
I guess one way to de-risk this is to have a separate `MaxSeatsTracker` that is updated in the controller loop with a write lock and then read with a separate read lock everytime. Thoughts?

Reply:
^ going to implement this, will ping you for another review later today when it's done.

Reply:
Updated, PTAL